### PR TITLE
Change the license to use the new endpoint and format

### PR DESCRIPTION
**Pull request by ph**
1. Use the `_license` endpoint instead of using the `/_xpack`
2. Remove the `features` handling.
3. Add more golden files for the different returned format.
4. Remove unused code.

Fixes: elastic/beats#15090

### CODE REVIEW

File: x-pack/libbeat/licenser/elastic_fetcher.go [hunk 1]
```diff
@@ -14,12 +14,11 @@ import (
 
 	"github.com/pkg/errors"
 
-	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
 )
 
-const xPackURL = "/_xpack"
+const xPackURL = "/_license"
```
Comment (urso) [inline x-pack/libbeat/licenser/elastic_fetcher.go]: hm... would it make sense to be BC compatible and set the URL based on the ES version reported by the ES Client?

**Comment by urso** [comment]
This PR removes quite some code. What is the motivation and actual impact of removing the license manager and watcher?

File: x-pack/libbeat/licenser/elastic_fetcher.go (hunk 1 above)
Comment (ph) [reply x-pack/libbeat/licenser/elastic_fetcher.go]: Both endpoint answer on  7.x series for Elastic Licensed ES so I don't think it's necessary to have it backward compatible. Also we only added this checin in 7.x.

**Comment by ph** [comment]
> This PR removes quite some code. What is the motivation and actual impact of removing the license manager and watcher?

Only functionbeat was actually using the Watcher/License manager and when we added Logstash back to the supported output we normalized the check to use the same between all the x-pack code which is just using a client calback. It requires one call but simplifies the logic on all the Elastic licensed beats. This move effectively made the Watcher and the Manager class unused/dead code. So I think it's better if we just remove it. If needed we can always ressurect it.

**Comment by ph** [comment]
testing license-related check is not a simple task when its not "trial" or "basic", in the linked private issues there are test servers that @bytebilly has created with credentials to be able to tested with a real server. The other way is to use the generated license in the private repository with dev artifact to run a local server. cc @fearful-symmetry

**Pull request merged**